sdks/router-sdk/src/entities/mixedRoute/route.ts

```diff
@@ -55,16 +62,30 @@ export class MixedRouteSDK<TInput extends Currency, TOutput extends Currency> {
       const pool = pools[i]
       const inputToken = tokenPath[i]
 
-      const outputToken =
-        pool instanceof V4Pool
-          ? pool.token0.equals(inputToken)
-            ? pool.token1
-            : pool.token0
-          : pool.token0.wrapped.equals(inputToken.wrapped)
-          ? pool.token1
-          : pool.token0
-
-      invariant(isValidTokenPath(prevPool, pool, inputToken), `PATH`)
+      let outputToken
+      if (
+        (pool instanceof V4Pool && !pool.involvesToken(inputToken)) ||
+        (!(pool instanceof V4Pool) && inputToken.isNative)
+      ) {
+        // We handle the case where the inputToken =/= pool.token0 or pool.token1. There are 2 specific cases.
+        if (inputToken.equals(pool.token0.wrapped)) {
+          // 1) the inputToken is WETH and the current pool has ETH
+          // path: USDC, WETH, PEPE
+          // pools: USDC-WETH, ETH-PEPE
+          outputToken = pool.token1
+        } else if (inputToken.wrapped.equals(pool.token0) || inputToken.wrapped.equals(pool.token1)) {
+          // 2) the inputToken is ETH and the current pool has WETH
+          // path: USDC, ETH, PEPE
+          // pools: USDC-ETH, WETH-PEPE
+          outputToken = inputToken.wrapped.equals(pool.token0) ? pool.token1 : pool.token0
+        } else {
+          throw new Error(`POOL_MISMATCH pool: ${JSON.stringify(pool)} inputToken: ${JSON.stringify(inputToken)}`)
+        }
+      } else {
+        // then the input token must equal either token0 or token1
+        invariant(isValidTokenPath(prevPool, pool, inputToken), `PATH`)
```

Comment:
`isValidTokenPath` doesnt check that theyre strictly equal - it involves wrapping checks too so will allow ETH and WETH to be "equal". So I dont think its the right function here? I think this line should just check

```suggestion
        invariant(inputToken.equals(pool.token0) || inputToken.equals(pool.token1), `PATH`)
```